Source/SourceKittenFramework/String+SourceKitten.swift

```diff
@@ -232,14 +232,15 @@ extension NSString {
     - parameter characterSet: Character set to check for membership.
     */
     public func trimmingTrailingCharacters(in characterSet: CharacterSet) -> String {
-        if length == 0 {
+        guard length > 0 else {
             return ""
         }
-        var charBuffer = [unichar](repeating: 0, count: length)
-        getCharacters(&charBuffer, range: NSRange(location: 0, length: charBuffer.count))
-        for newLength in (1...length).reversed() {
-            if !characterSet.contains(UnicodeScalar(charBuffer[newLength - 1])!) {
-                return substring(with: NSRange(location: 0, length: newLength))
+        var string = self.bridge()
+        let endIndex = string.unicodeScalars.endIndex
+        for offset in 1...string.unicodeScalars.count {
+            let index = string.unicodeScalars.index(endIndex, offsetBy: -offset)
```

Comment:
Yes, you're right. Thanks for the input. I think there's even a much easier implementation (at least for me it's much easier to wrap my head around 😉 ):
```Swift
        var unicodeScalars = self.bridge().unicodeScalars
        while let scalar = unicodeScalars.last {
            if !characterSet.contains(scalar) {
                return String(unicodeScalars)
            }
            unicodeScalars.removeLast()
        }
```

Do you have concerns?